src/main/scala/uk/gov/nationalarchives/consignmentexport/ChecksumValidator.scala

```diff
@@ -0,0 +1,32 @@
+package uk.gov.nationalarchives.consignmentexport
+
+import java.util.UUID
+
+import cats.effect.IO
+import gov.loc.repository.bagit.domain.Bag
+import io.chrisdavenport.log4cats.SelfAwareStructuredLogger
+import uk.gov.nationalarchives.consignmentexport.Validator.ValidatedFileMetadata
+
+import scala.jdk.CollectionConverters._
+
+class ChecksumValidator(consignmentId: UUID)(implicit val logger: SelfAwareStructuredLogger[IO]) {
+
+  def validateFileChecksums(bag: Bag,
+                            validatedFileMetadata: List[ValidatedFileMetadata]): Either[RuntimeException, IO[Unit]] = {
+
+    val bagitGeneratedChecksums = bag.getPayLoadManifests.asScala.head.getFileToChecksumMap.values()
```

Comment:
Just to check: does this do any IO operations, or is reading all the information from memory?

Reply:
To be honest I don't know